### PR TITLE
Moved TriggerType enum to dedicated file to avoid repetition in hardware component classes

### PR DESCRIPTION
**Pull request by kumar-sanjeeev**
- Moved `TriggerType` enum to `hardware_interface_trigger_type.hpp`, to avoid repetition across hardware components classes.  

Fixes: https://github.com/ros-controls/ros2_control/issues/1958

### CODE REVIEW

**Review by saikishor — COMMENTED** [review]
May be we can add it to the `hardware_interface_return_values.hpp`  along side  https://github.com/ros-controls/ros2_control/blob/612b30ba655922c51e7a07e077d3f3ca06ea2092/hardware_interface/include/hardware_interface/types/hardware_interface_return_values.hpp#L20-L29 instead of creating a new one.

WHat's your opinion on this @ros-controls/ros2-maintainers ?

**Comment by bmagyar** [comment]
Looks good to me. I'd simplify the file name to "trigger_type.hpp", the "hardware_interface" is already implied

File: hardware_interface/include/hardware_interface/types/hardware_interface_trigger_type.hpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
```
Comment (bmagyar) [inline hardware_interface/include/hardware_interface/types/hardware_interface_trigger_type.hpp:1]:
```suggestion
// Copyright 2024 ros2_control Development Team
```
just a small adjustment here, you can apply/commit the change directly

**Review by christophfroehlich — APPROVED** [review]
Thanks!

**Review by saikishor — APPROVED** [review]

**Pull request merged**